flexmeasures/data/schemas/generic_assets.py

```diff
@@ -42,32 +42,26 @@ class GenericAssetSchema(ma.SQLAlchemySchema):
     longitude = LongitudeField(allow_none=True)
     generic_asset_type_id = fields.Integer(required=True)
     attributes = JSON(required=False)
+    parent_asset_id = fields.Int(required=False, allow_none=True)
 
     class Meta:
         model = GenericAsset
 
     @validates_schema(skip_on_field_errors=False)
-    def validate_name_is_unique_in_account(self, data, **kwargs):
+    def validate_name_is_unique_under_parent(self, data, **kwargs):
         if "name" in data:
-            if data.get("account_id") is None:
-                asset = GenericAsset.query.filter(
-                    GenericAsset.name == data["name"], GenericAsset.account_id.is_(None)
```

Comment:
Here we had a diverging syntax to check for None values (public account). Is that no longer needed?

Reply:
Good point!

The use of the function `_is` is no longer needed as SQLAlchemy is using the right expression for the resulting SQL query:

With `account_id is None`:

```sql
SELECT generic_asset.id AS generic_asset_id, generic_asset.name AS generic_asset_name, generic_asset.latitude AS generic_asset_latitude, generic_asset.longitude AS generic_asset_longitude, generic_asset.attributes AS generic_asset_attributes, generic_asset.parent_asset_id AS generic_asset_parent_asset_id, generic_asset.generic_asset_type_id AS generic_asset_generic_asset_type_id, generic_asset.account_id AS generic_asset_account_id
FROM generic_asset
WHERE generic_asset.name = %(name_1)s AND generic_asset.parent_asset_id IS NULL AND generic_asset.account_id = %(account_id_1)s
```

With `account_id is not None`:

```sql
SELECT generic_asset.id AS generic_asset_id, generic_asset.name AS generic_asset_name, generic_asset.latitude AS generic_asset_latitude, generic_asset.longitude AS generic_asset_longitude, generic_asset.attributes AS generic_asset_attributes, generic_asset.parent_asset_id AS generic_asset_parent_asset_id, generic_asset.generic_asset_type_id AS generic_asset_generic_asset_type_id, generic_asset.account_id AS generic_asset_account_id
FROM generic_asset
WHERE generic_asset.name = %(name_1)s AND generic_asset.parent_asset_id IS NULL AND generic_asset.account_id = %(account_id_1)s
```

Reply:
According to [[1]](https://stackoverflow.com/a/5632224), the `is_` operator was introduced on SQLAlchemy 0.7.9. Prior to that, SQLAlchemy already transformed `==` into `=` or `IS` depending on the context.

Currently, we are using the following constraint for SQLAlchemy:

```
sqlalchemy>=1.4.0, <2
```

We are good to go I think.


[1] https://stackoverflow.com/a/5632224